langchain/llms/azureml_endpoint.py

```diff
@@ -0,0 +1,238 @@
+from langchain.utils import get_from_dict_or_env
+
+INPUT_TYPE = TypeVar("INPUT_TYPE", bound=Union[str, List[str]])
+OUTPUT_TYPE = TypeVar("OUTPUT_TYPE", bound=Union[str, List[List[float]]])
+
+
+class AzureMLEndpointClient(object):
+    """Wrapper around AzureML Managed Online Endpoint Client."""
+
+    def __init__(
+        self, endpoint_url: str, endpoint_api_key: str, deployment_name: str
+    ) -> None:
+        """Initialize the class."""
+        self.endpoint_url = endpoint_url
+        self.endpoint_api_key = endpoint_api_key
+        self.deployment_name = deployment_name
+
+    def call(self, body: bytes) -> Union[Dict, str]:
+        """call."""
+
+        url = self.endpoint_url
+        # Replace this with the primary/secondary key or AMLToken for the endpoint
+        api_key = self.endpoint_api_key
+        if not api_key:
+            raise Exception("A key should be provided to invoke the endpoint")
+
+        # The azureml-model-deployment header will force the request to go to a
+        # specific deployment. Remove this header to have the request observe the
+        # endpoint traffic rules.
+
+        headers = {
+            "Content-Type": "application/json",
+            "Authorization": ("Bearer " + api_key),
+            "azureml-model-deployment": self.deployment_name,
+        }
+
+        req = urllib.request.Request(url, body, headers)
+        try:
+            response = urllib.request.urlopen(req, timeout=50)
+            result = response.read()
+        except urllib.error.HTTPError as error:
+            print("The request failed with status code: " + str(error.code))
+            result = str(error)
+        except Exception as e:
+            print("Calling Azure Managed Online endpoint failed!")
+            result = str(e)
+        return result
+
+
+class ContentFormatterBase(Generic[INPUT_TYPE, OUTPUT_TYPE]):
+    """A handler class to transform request and response of
+    AzureML endpoint to match with required schema.
+    """
+
+    """
+    Example:
+        .. code-block:: python
+        
+            class ContentFormatter(ContentFormatterBase[str, str]):
+                content_type = "application/json"
+                accepts = "application/json"
+                
+                def format_request_payload(
+                    self, 
+                    prompt: str, 
+                    model_kwargs: Dict
+                ) -> bytes:
+                    input_str = json.dumps(
+                        {
+                            "inputs": {"input_string": [prompt]}, 
+                            "parameters": model_kwargs,
+                        }
+                    )
+                    return str.encode(input_str)
+                    
+                def format_response_payload(self, output: str) -> str:
+                    response_json = json.loads(output)
+                    return response_json[0]["0"]
+    """
+    content_type: Optional[str] = "text/plain"
```

Comment:
should defaults be application/json for contrent_type and accepts, if that's what all current child classes use?

Reply:
Yes, that makes sense